### PR TITLE
executor: remove joinResultGenerator.emit and rename emitToChunk to emit

### PR DESCRIPTION
**Pull request by XuHuaiyu**
As the title says.
PTAL @zz-jason @winoros

### CODE REVIEW

File: executor/join_result_generators.go [hunk 1]
```diff
@@ -35,30 +35,26 @@ var (
 // joinResultGenerator is used to generate join results according the join type, see every implementor for detailed information.
 type joinResultGenerator interface {
 	// emit tries to join an outer row with a batch of inner rows.
-	// When len(inners) == 0, it means that the outer row can not be joined with any inner row:
+	// When inners == nil or inners.Len() == 0, it means that the outer row can not be joined with any inner row:
```
Comment (zz-jason) [inline executor/join_result_generators.go]: `inners == nil` is covered by `len(inners) == 0`

Comment (XuHuaiyu) [reply executor/join_result_generators.go]: It's `inners.Len`, not `len(inners)`.

**Comment by zz-jason** [comment]
LGTM

**Pull request merged**